graphql_compiler/macros/macro_edge/helpers.py

```diff
@@ -53,6 +60,191 @@ def get_directives_for_ast(ast):
+    """
+    return {
+        # Schema validation has ensured this exists
+        directive.arguments[0].value.value
+        for ast, directive in _yield_ast_nodes_with_directives(ast)
+        if directive.name.value == TagDirective.name
+    }
+
+
+def _replace_tag_names_in_tag_directive(name_change_map, tag_directive):
+    """Return the new directive, and whether it is different from the old.
+
+    Args:
+        name_change_map: Dict[str, str] mapping tag names to new names
+        tag_directive: GraphQL library tag directive whose name is in the name_change_map.
+                       This ast is not mutated.
+
+    Returns:
+        tuple containing:
+        - GraphQL library directive, equivalent to the input one, with its name changed
+          according to the name_change_map
+        - bool, whether the output is different from the input
+    """
+    # Schema validation has ensured this exists
+    current_name = tag_directive.arguments[0].value.value
+    new_name = name_change_map[current_name]
+    renamed_tag_directive = copy(tag_directive)
+    renamed_tag_directive.arguments = [Argument(Name('tag_name'), StringValue(new_name))]
+    return renamed_tag_directive, new_name != current_name
+
+
+def _replace_tag_names_in_filter_directive(name_change_map, filter_directive, ast):
+    """Return the new directive, and whether it is different from the old.
+
+    Args:
+        name_change_map: Dict[str, str] mapping tag names to new names
+        filter_directive: GraphQL library filter directive that potentially uses tagged parameters.
+                          all such tagged parameters should be in the name_change_map. This
+                          directive is not mutated.
+
+    Returns:
+        tuple containing:
+        - GraphQL library directive, equivalent to the input one, with any tagged parameters it
+          uses replaced according to the name_change_map
+        - bool, whether the output is different from the input
+    """
+    made_changes = False
+    filter_with_renamed_args = copy(filter_directive)
+    filter_with_renamed_args.arguments = []
+    for argument in filter_directive.arguments:
+        if argument.name.value == 'op_name':
+            filter_with_renamed_args.arguments.append(argument)
+        elif argument.name.value == 'value':
+            new_value_list = []
+            for value in argument.value.values:
+                if is_tagged_parameter(value.value):
+                    current_name = get_parameter_name(value.value)
+                    new_name = name_change_map[current_name]
+                    if new_name != current_name:
+                        made_changes = True
+                    new_value_list.append(StringValue('%' + new_name))
+                elif is_runtime_parameter(value.value):
+                    new_value_list.append(value)
+                else:
+                    raise AssertionError(u'Value was neither tag nor filter: {} {}'
```

Comment:
This error message is confusing. `Value` is vague, and `tag nor filter` makes it sound like the problem is with the directive. Let's stick to the naming conventions we've adopted in the README: `runtime parameter` and `tagged parameter`.